code/game/objects/items/devices/scanners.dm

```diff
@@ -361,8 +363,40 @@ GAS ANALYZER
 	user.visible_message("<span class='suicide'>[user] begins to analyze [user.p_them()]self with [src]! The display shows that [user.p_theyre()] dead!</span>")
 	return BRUTELOSS
 
-/obj/item/analyzer/attack_self(mob/user)
+/obj/item/analyzer/afterattack(obj/holder, mob/user, proximity_flags)
+	if(holder.Adjacent(user))
+		//there has to be a better way to do this
+		if (istype(holder, /obj/item/tank))
+			var/obj/item/tank/T = holder
+			atmosanalyzer_scan(T.air_contents, user, T)
+
+		else if (istype(holder, /obj/item/onetankbomb))

```

Comment:
This is so gross. Why not have `analyzer_act`?

Reply:
i think its ugly too, i just dont know an objectively better way to do it
care to elaborate?

Reply:
actually I think I know what you mean
i'll try that out

Reply:
@Fox-McCloud changed, this is what you meant yes?

Reply:
Yes, it is. Thank you so much. This is much more OOP and far superior. Really appreciate it.